### ISSUE TITLE
Bug on simple rounding

### ISSUE THREAD

**Issue by adshrc** [josdejong/mathjs#3100]
The issues section is used only for bug reports. Please use the [Discussions](https://github.com/josdejong/mathjs/discussions) section to ask questions and share ideas and suggestions.

**Describe the bug**
I use this library for simple rounding of currency values. Below, I want to convert 0.145 € to cents.

**To Reproduce**
round(0.145*100) returns 14, which should be actually 15
round(evaluate('0.145 * 100')) returns also 14, which is wrong
round(14.5) returns 15, which is correct

**Comment by rakheesingh** [comment]
@josdejong I wanted to take this this issue.

**Comment by josdejong** [comment]
Thanks @rakheesingh! Please let me know when you need help (I haven't looked into this and I'm not sure what is needed to fix it).

**Comment by developer-diganta** [comment]
Hello @josdejong! Wanted to know if the issue is still open? Would love to give it a try!

**Comment by adshrc** [comment]
adding an epsilon fixes it:

```
round(number: number, decimals = 0): number {
    // just using the bare round function is not a good idea: https://github.com/josdejong/mathjs/issues/3100
    // to prevent this error, add an epsilon
    const epsilon = 1e-12;
    return round(number + epsilon, decimals);
  }
```

**Comment by josdejong** [comment]
The issue is still open, thanks @developer-diganta for wanting to look into it. Help would be welcome.

**Comment by josdejong** [comment]
Fixed now in `v12.4.0` via #3136.
